src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -32,13 +31,13 @@ function CompleteBipartiteGraph(n1::T, n2::T) where {T <: Integer}
     (n1 < 0 || n2 < 0) && return SimpleGraph{T}(0)
     Tw = widen(T)
     nw = Tw(n1) + Tw(n2)
-    n = T(n1 + n2)  # checks if T is large enough for n1 + n2
+    n = T(nw)  # checks if T is large enough for n1 + n2
 
     ne = Int(n1) * Int(n2)
 
     fadjlist = Vector{Vector{T}}(undef, n)
-    range1 = UnitRange{T}(1, n1)
-    range2 = UnitRange{T}(n1 + 1, n)
+    range1 = 1:n1
+    range2 = (n1 + 1):n
```

Comment:
Why do we no longer care that the ranges have eltype `T`?

```
julia> a = Int8(5)
5

julia> 1:a
1:5

julia> typeof(1:a)
UnitRange{Int64}
```

Reply:
Same reason as above and in addition the range here is also used for indexing a Vector where `Int` is used anyway.